test/types/records/generic/dispatchWithinIntParamLoop.chpl

```diff
@@ -0,0 +1,44 @@
+record R {
+  param flag = true;
+}
+
+proc foo(x: R) {
+  writeln("In foo: ", x.type:string);
+}
+
+proc bar(x: R(true)) {
+  writeln("In bar: ", x.type:string);
+}
+
+proc bar(x: R(false)) {
+  writeln("In bar: ", x.type:string);
+}
+
+proc baz(x: R(?)) {
+  writeln("In baz: ", x.type:string);
+}
+
+// Works:
+
+{
+  var r = new R(false);
+  bar(r);
+  baz(r);
+}
+
+{
+  var r = new R(true);
+  foo(r);
+  bar(r);
+  baz(r);
+}
+
+// Doesn't work
+
+for param flag in 0..1 {
+  var r = new R(flag:bool);
+  if flag:bool == true then foo(r);
+  bar(r);
+  baz(r);
+}
+
```

Comment:
When the loop is a param int loop it works; only the param bool loop doesn't.  Or didn't… Daniel fixed it since I opened the issue.  This filename is slightly different than the one named there:  dispatchWithinParamLoop.chpl

Reply:
Oops, yes, I didn't notice that and should've removed it.  I am so bad at reading comments, even my own…